R/github.R

```diff
@@ -66,26 +66,29 @@ use_github <- function(organisation = NULL,
                        auth_token = NULL,
                        host = NULL) {
   check_uses_git()
+  check_branch("master")
+  check_uncommitted_changes()
+  check_no_origin()
```

Comment:
I think aborting if origin is specified is too strong - we should be able to complete the branch set up if they got the remote set up last time.

Reply:
I really think we shouldn't touch a repo that has a preconfigured `origin` remote. We do say exactly how to remove this, so they can call `use_github()` again.

If we accept a pre-existing `origin`, now we have to see where it points. Is it a GitHub URL? Does that repo already exist? Does it have the owner and repo name we expect? Etc etc.